storage/drive/drive.go

```diff
@@ -9,26 +9,69 @@ import (
 	"os"
 	"strings"
 
-	"github.com/go-kit/log"
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/mapstructure"
 	"google.golang.org/api/drive/v3"
+	"google.golang.org/api/option"
 
 	"github.com/connylabs/ingest"
+	iplugin "github.com/connylabs/ingest/plugin"
 	"github.com/connylabs/ingest/storage"
 )
 
-type driveStorage struct {
+type DriveStorage struct {
 	s *drive.Service
-	l log.Logger
+	l hclog.Logger
 	p string
 }
 
+type destinationConfig struct {
+	APIKey          string
+	CredentialsFile string
+	Folder          string
+}
+
+// NewDestination implements the Plugin interface.
+func (p *DriveStorage) Configure(config map[string]interface{}) error {
```

Comment:
I'm a bit confused. Why are we trying to implement the Plugin interface here in this package? The S3 plugin, for example, only implements the `Configure` method in the `plugins/s3` package. I would think we would do the same for the drive plugin and this package would remain a place to implement the `Storage` interface.

Reply:
I think I would rather get rid of the Storage interface and use only the Destination interface. They are almost the same.

I moved the Configure method here so I don't have to implement it in the main package in plugins/drive/main.go. It would make testing more difficult because we can not import an implementation of the Destination interface for drive in other packages.

Reply:
I just don't think it makes sense to describe a `Configure` method at all for storage. It's obviously a plugin-related concept and we are being disingenuous /lazy by putting it here. Maybe we want to move everything somewhere else? Like put _all_ of this under `plugin/destination` and `plugin/source`? It's also odd and inconsistent that half of the S3 plugin is implemented in one place and the other half is implemented in the other place. The implementation really _should_ live in one place. The fact that we are fulfilling our needs by using two separate plugins to implement each plugin is kind of a side-effect and shouldn't be dictating how we structure the code-base IMO. S3 is one (meta-)plugin and Drive should be one plugin and the plugins should live in one consistent place IMO

Reply:
yes, so let's move all the plugin implementation to plugins/... and move their main packages to cmd/plugins or something?

And we remove the whole storage package?